imagededup/utils/general_utils.py

```diff
@@ -65,3 +66,20 @@ def parallelise(function: Callable, data: List, verbose: bool) -> List:
     pool.close()
     pool.join()
     return results
+
+
+def generate_files(image_dir: Union[PurePath, str], recursive: bool) -> List:
+    if recursive:
+        glob = "**/*"
+    else:
+        glob = "*"
+
+    return [
+        i.absolute()
+        for i in Path(image_dir).glob(glob)
+        if not (i.name.startswith(".") or i.is_dir())
+    ]
+
+
+def generate_relative_names(image_dir: Union[PurePath, str], files: List) -> List:
+    return [f.relative_to(Path(image_dir)).as_posix() for f in files]
```

Comment:
Use of `as_posix()` will make it difficult for windows users who will get forward slashes as separators. I think it is best to ignore inter-platform operability for now.
```suggestion
    return [str(f.relative_to(Path(image_dir)) for f in files]
```